modules/core/src/lib/deck-picker.js

```diff
@@ -26,31 +26,122 @@ import {getClosestObject, getUniqueObjects} from '../utils/query-object';
 import {processPickInfo, getLayerPickingInfo} from '../utils/pick-info';
 
 export default class DeckPicker {
-  constructor(props) {}
+  constructor(props) {
+    this.gl = props.gl;
+    this.layerManager = props.layerManager;
```

Comment:
Add a TODO to remove the layer manager reference. I see that as an important step 3 of this refactor.